cmd/compose/convert.go

```diff
@@ -139,15 +137,10 @@ func runConvert(ctx context.Context, streams api.Streams, backend api.Service, o
 		return nil
 	}
 
-	var out io.Writer = streams.Out()
 	if opts.Output != "" && len(content) > 0 {
```

Comment:
The only thing I'm looking at here (but this was already the case); should `let(content) == 0` be an error condition? In what situation would an empty content be created, and should that be an error? 

Currently, the code skips creating the file (which is good), but then continues to writing "nothing" to stdout; maybe it should be an error if there's no content created 🤔 

(Probably better for maintainers to have a look at, and likely better for a follow-up)

Reply:
Good point! The code would cleaner, although I don't think the content could be empty without throwing an error before that line. The content of the compose file is validated before that, so even if I have an empty compose file, that would be invalid, therefore it couldn't reach the line of that check.  If I have a valid compose file, even if it doesn't have any services only a mapping that could be used for anchors like this:

```yaml
x-test:
  hello: world
```
It would result a longer and not shorter output:
```yaml
name: phptest
services: {}
x-test:
  hello: world
```
Even if I use `--format json`, the project name and the empty services mapping would be generated.

If there is a way we don't know about to have an empty output without error messages, simply just removing `&& len(content) > 0` could be a good way to solve this, so you could have an empty file as you would expect it when you have a valid empty output.

Of course other parts of the code could be changed later to eventually generate an empty and valid output, but I agree, this is something that the maintainers should decide how they want to handle it.

Again, thanks for pointing it out :) I enjoy learning about the code of Docker Compose.